device_authorize_request_handler.go

```diff
@@ -0,0 +1,37 @@
+// Copyright © 2023 Ory Corp
+// SPDX-License-Identifier: Apache-2.0
+
+package fosite
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/ory/fosite/i18n"
+	"github.com/ory/x/errorsx"
+)
+
+func (f *Fosite) NewDeviceAuthorizeRequest(ctx context.Context, r *http.Request) (DeviceAuthorizeRequester, error) {
+	request := NewDeviceAuthorizeRequest()
+	request.Lang = i18n.GetLangFromRequest(f.Config.GetMessageCatalog(ctx), r)
+
+	if err := r.ParseForm(); err != nil {
+		return nil, errorsx.WithStack(ErrInvalidRequest.WithHint("Unable to parse HTTP body, make sure to send a properly formatted form request body.").WithWrap(err).WithDebug(err.Error()))
```

Comment:
I don't really know why, but i know that `AuthorizeRequestHandler` is doing the same:
https://github.com/ory/fosite/blob/master/authorize_request_handler.go#L325

Reply:
I thought that for authorization code, this was because of PAR (https://datatracker.ietf.org/doc/html/rfc9126)? So that endpoint actually supports POST. Right?

Reply:
I've refactor the way I'm querying the QueryParams. I now use `r.URL.Query()`